src/opt/ReplaceMaskedMemOps.cpp

```diff
@@ -99,6 +99,21 @@ llvm::Constant *lExtractSecondPartOfConstVec(llvm::LLVMContext &context, llvm::C
     return llvm::ConstantVector::get(newValuesRef);
 }
 
+// Extend the vector of length 3 to the vector of length 4. The extra element
+// is taken from the second vector (the first element).
+llvm::Value *lExtendVec3ToVec4(llvm::IRBuilder<> &B, llvm::Value *vec3, llvm::Value *sec, const llvm::Twine &name) {
+    auto *vec3Type = llvm::dyn_cast<llvm::VectorType>(vec3->getType());
+    Assert(vec3Type && vec3Type->getElementCount().getKnownMinValue() == 3);
+
+    std::vector<unsigned> indices3 = {0, 0, 0};
+    llvm::Constant *mask3 = llvm::ConstantDataVector::get(B.getContext(), indices3);
+    llvm::Value *sec3 = B.CreateShuffleVector(sec, sec, mask3);
```

Comment:
Let's add Assert checking that `sec` has a VectorType

Reply:
Done